subgraphs/hop-protocol/protocols/hop-protocol/config/deployments/hop-protocol-arbitrum/configurations.ts

```diff
@@ -0,0 +1,302 @@
+import { log } from "@graphprotocol/graph-ts";
+import { Configurations } from "../../../../../configurations/configurations/interface";
+import {
+  ArbitrumToken,
+  MainnetToken,
+  ArbitrumAmm,
+  ArbitrumBridge,
+  XdaiToken,
+  PolygonToken,
+  OptimismToken,
+  ArbitrumRewardToken,
+  ZERO_ADDRESS,
+  RewardTokens,
+  ArbitrumHtoken,
+} from "../../../../../src/sdk/util/constants";
+import { Network } from "../../../../../src/sdk/util/constants";
+export class HopProtocolArbitrumConfigurations implements Configurations {
+  getNetwork(): string {
+    return Network.ARBITRUM_ONE;
+  }
+
+  getPoolAddressFromTokenAddress(tokenAddress: string): string {
+    if (tokenAddress == ArbitrumToken.USDC) return ArbitrumAmm.USDC;
+    else if (tokenAddress == ArbitrumToken.DAI) return ArbitrumAmm.DAI;
+    else if (tokenAddress == ArbitrumToken.USDT) return ArbitrumAmm.USDT;
+    else if (tokenAddress == ArbitrumToken.ETH) return ArbitrumAmm.ETH;
+    else if (tokenAddress == ArbitrumToken.rETH) return ArbitrumAmm.rETH;
+    else {
+      log.critical("Token not found", []);
+      return "";
+    }
+  }
+  getTokenDetails(tokenAddress: string): string[] {
+    if (this.getUsdcTokens().includes(tokenAddress)) {
+      return ["USDC", "USDC", "6", ArbitrumBridge.USDC];
+    } else if (this.getDaiTokens().includes(tokenAddress)) {
+      return ["DAI", "DAI", "18", ArbitrumBridge.DAI];
+    } else if (this.getUsdtTokens().includes(tokenAddress)) {
+      return ["USDT", "USDT", "6", ArbitrumBridge.USDT];
+    } else if (this.getRethTokens().includes(tokenAddress)) {
+      return ["rETH", "Rocket Pool Ethereum", "18", ArbitrumBridge.rETH];
+    } else if (this.getEthTokens().includes(tokenAddress)) {
+      return ["ETH", "ETH", "18", ArbitrumBridge.ETH];
+    } else if (tokenAddress == RewardTokens.GNO) {
+      return ["GNO", "Gnosis Token", "18", ZERO_ADDRESS];
+    } else if (tokenAddress == RewardTokens.rETH_ARB) {
+      return ["rETH", "Rocket Pool Ethereum", "18", ZERO_ADDRESS];
+    } else if (tokenAddress == RewardTokens.HOP) {
+      return ["HOP", "HOP Token", "18", ZERO_ADDRESS];
+    } else {
+      log.critical("Token not found", []);
+      return [];
+    }
+  }
+
+  getCrossTokenAddress(chainId: string, tokenAddress: string): string {
+    if (chainId == "42161")
+      return this.getArbitrumCrossTokenFromTokenAddress(tokenAddress);
+    else if (chainId == "10")
+      return this.getOptimismCrossTokenFromTokenAddress(tokenAddress);
+    else if (chainId == "100")
+      return this.getXdaiCrossTokenFromTokenAddress(tokenAddress);
+    else if (chainId == "137")
+      return this.getPolygonCrossTokenFromTokenAddress(tokenAddress);
+    else if (chainId == "1")
+      return this.getMainnetCrossTokenFromTokenAddress(tokenAddress);
+    else {
+      log.critical("Chain not found", []);
+      return "";
+    }
+  }
+
+  getArbitrumCrossTokenFromTokenAddress(tokenAddress: string): string {
```

Comment:
Is this function necessary if it doesn't do anything?

Reply:
Yeah, that function is only used on L1. All the configs share the same interface. 
Removing that throws an error that we need the function is missing IIRC

Reply:
Can you make a comment in the code as to why we need it then?

Reply:
Okay, I'll add those.

Reply:
@dmelotik I opted to just return the tokenAddress in the function even though it will never be called. 
Does this solution work? If not I can revert. 
Leaving a comment means I'd have repeat the same comments a lot.

Reply:
I am not sure if typescript allows optional functions on an interface, but another option is to extend the interface.

If it works as is though, we can leave it.